android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java

```diff
@@ -166,6 +174,88 @@ private void connectRealtime(@NonNull MethodCall call, @NonNull MethodChannel.Re
         });
     }
 
+    private void attachRealtimeChannel(
+            @NonNull MethodCall call, @NonNull MethodChannel.Result result
+    ) {
+        final AblyFlutterMessage message = (AblyFlutterMessage)call.arguments;
+        this.<AblyFlutterMessage<Map<String, Object>>>ablyDo(message, (ablyLibrary, ablyMessage) -> {
+            try {
+                String channelName = (String) ablyMessage.message.get("channel");
+                ChannelOptions channelOptions = (ChannelOptions) ablyMessage.message.get("options");
+                ablyLibrary
+                        .getRealtime(ablyMessage.handle)
+                        .channels
+                        .get(channelName, channelOptions)
+                        .attach(
+                                new CompletionListener() {
+                                    @Override
+                                    public void onSuccess() {
+                                        result.success(null);
+                                    }
+
+                                    @Override
+                                    public void onError(ErrorInfo reason) {
+                                        handleAblyException(result, AblyException.fromErrorInfo(reason));
+                                    }
+                                }
+                        );
+            }catch(AblyException e){
+                handleAblyException(result, e);
+            }
+        });
+    }
+
+    private void detachRealtimeChannel(
+            @NonNull MethodCall call, @NonNull MethodChannel.Result result
+    ) {
+        final AblyFlutterMessage message = (AblyFlutterMessage)call.arguments;
+        this.<AblyFlutterMessage<Map<String, Object>>>ablyDo(message, (ablyLibrary, ablyMessage) -> {
+            try {
+                String channelName = (String) ablyMessage.message.get("channel");
+                ChannelOptions channelOptions = (ChannelOptions) ablyMessage.message.get("options");
+                ablyLibrary
+                        .getRealtime(ablyMessage.handle)
+                        .channels
+                        .get(channelName, channelOptions)
+                        .detach(
+                                new CompletionListener() {
+                                    @Override
+                                    public void onSuccess() {
+                                        result.success(null);
+                                    }
+
+                                    @Override
+                                    public void onError(ErrorInfo reason) {
+                                        handleAblyException(result, AblyException.fromErrorInfo(reason));
+                                    }
+                                }
+                        );
+            }catch(AblyException e){
+                handleAblyException(result, e);
```

Comment:
Noticing quite a lot of repetition with these `catch` blocks in every closure passed to `ablyDo` which makes me wonder if `ablyDo` couldn't catch `AblyException` instances itself, accepting the `result` object itself so it can deal with them without all this boilerplate. What do you think?

Reply:
This requires us to have a custom biconsumer which can throw as the default `BiConsumer~accept` doesn't throw.
But that could lead to some unwanted as AblyException is not always thrown and we are using `ablyDo` for all the method calls.

Reply:
Fair enough. Thanks for explaining.